src/test/java/net/openhft/chronicle/wire/method/GenerateMethodWriterInheritanceTest.java

```diff
@@ -102,14 +103,16 @@ public void testGenerateForClass() {
     public void testGenerateForLongGeneratedClassName() {
         final Wire wire = BINARY.apply(Bytes.allocateElasticOnHeap());
 
-        wire.methodWriter(
+        Object writer = wire.methodWriter(
                 NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener1.class,
                 NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener2.class,
                 NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener3.class
         );
+        assertInstanceOf(NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener1.class, writer);
+        assertInstanceOf(NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener2.class, writer);
+        assertInstanceOf(NewOrderSingleListenerOmsHedgerTradeListenerOpenOrdersListenerPaidGivenTickListener3.class, writer);
```

Comment:
I pushed a change to use `assertInstanceOf(x, y)` instead of `assertTrue(x instanceof y)` as it allows JUnit to give a better error message when it fails.